### PR TITLE
chore: update csi-secrets-store addon manifest and images (v0.0.9)

### PR DESCRIPTION
**Pull request by jadarsie**
**Reason for Change**:

- Update csi-secrets-store addon manifests for latest release (v0.0.9)
- Update images in VHD

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/addons/secrets-store-csi-driver.yaml [hunk 1]
```diff
@@ -451,16 +458,26 @@ spec:
         app: csi-secrets-store-provider-azure
     spec:
       serviceAccountName: csi-secrets-store-provider-azure
+      hostNetwork: true
```
Comment (aramase) [inline parts/k8s/addons/secrets-store-csi-driver.yaml:461]: So `0.0.9` for provider is not compatible with `v0.0.14` release of the driver. The following flag `--grpc-supported-providers=azure` needs to be added to the `secrets-store` container for the driver-provider communication to work. 

Ref doc - 
https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/how-to-upgrade.md

`--grpc-supported-providers=azure` needs to be added in [here](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/deploy/secrets-store-csi-driver.yaml#L53-L58)

**Review by aramase — CHANGES_REQUESTED** [review]

File: parts/k8s/addons/secrets-store-csi-driver.yaml (hunk 1 above)
Comment (aramase) [reply parts/k8s/addons/secrets-store-csi-driver.yaml:461]: FYI, for users using helm chart to deploy the driver and provider we configure it automatically - https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/charts/csi-secrets-store-provider-azure/values.yaml#L75

**Review by aramase — COMMENTED** [review]
lgtm

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**